### ISSUE TITLE
False positive `react/no-unescaped-entities` on whitespace with newlines inside JSX.

### ISSUE THREAD

**Issue by sompylasar** [eslint/typescript-eslint-parser#278]
**What version of TypeScript are you using?**
`2.3.2`

**What version of `typescript-eslint-parser` are you using?**
`3.0.0`

**What code were you trying to parse?**
Code is closed-source, but the gist is this:
```jsx
<SomeComponent>
  {children}
</SomeComponent>
```

**What did you expect to happen?**
No `react/no-unescaped-entities` warning.

**What happened?**
False positive `react/no-unescaped-entities` on whitespace with newlines inside JSX.
https://github.com/yannickcr/eslint-plugin-react/blob/f0dcaca92418bb791f585e867d720a96dd43a277/lib/rules/no-unescaped-entities.js#L72

The code gets parsed as:
```yaml
JSXElement: '<SomeComponent>'
  Literal: '\n  '
    ...
```
and `react/no-unescaped-entities` tells me to escape something there, likely the newline character.
After converting the code to:
```jsx
<SomeComponent>{children}</SomeComponent>
```
the warning goes away.

Originally submitted here: https://github.com/eslint/typescript-eslint-parser/issues/266#issuecomment-302839595

**Comment by sompylasar** [comment]
Refs https://github.com/eslint/typescript-eslint-parser/pull/277

**Comment by soda0289** [comment]
Thanks.
